src/consensus/aft/test/driver.h

```diff
@@ -74,59 +109,130 @@ class RaftDriver
 
   void log(ccf::NodeId first, ccf::NodeId second, const std::string& message)
   {
-    std::cout << "  Node" << first << "->>"
-              << "Node" << second << ": " << message << std::endl;
+    RAFT_DRIVER_OUT << "  Node" << first << "->>"
+                    << "Node" << second << ": " << message << std::endl;
   }
 
   void rlog(ccf::NodeId first, ccf::NodeId second, const std::string& message)
   {
-    std::cout << "  Node" << first << "-->>"
-              << "Node" << second << ": " << message << std::endl;
+    RAFT_DRIVER_OUT << "  Node" << first << "-->>"
+                    << "Node" << second << ": " << message << std::endl;
   }
 
   void log_msg_details(
     ccf::NodeId node_id, ccf::NodeId tgt_node_id, aft::RequestVote rv)
   {
-    std::ostringstream s;
-    s << "request_vote t: " << rv.term << ", lci: " << rv.last_committable_idx
-      << ", tolci: " << rv.term_of_last_committable_idx;
-    log(node_id, tgt_node_id, s.str());
+    const auto s = fmt::format(
+      "request_vote for term {}, at tx {}.{}",
+      rv.term,
+      rv.term_of_last_committable_idx,
+      rv.last_committable_idx);
+    log(node_id, tgt_node_id, s);
   }
 
   void log_msg_details(
     ccf::NodeId node_id, ccf::NodeId tgt_node_id, aft::RequestVoteResponse rv)
   {
-    std::ostringstream s;
-    s << "request_vote_response t: " << rv.term << ", vg: " << rv.vote_granted;
-    rlog(node_id, tgt_node_id, s.str());
+    const auto s = fmt::format(
+      "request_vote_response for term {} = {}",
+      rv.term,
+      (rv.vote_granted ? "Y" : "N"));
+    rlog(node_id, tgt_node_id, s);
   }
 
   void log_msg_details(
     ccf::NodeId node_id, ccf::NodeId tgt_node_id, aft::AppendEntries ae)
   {
-    std::ostringstream s;
-    s << "append_entries i: " << ae.idx << ", t: " << ae.term
-      << ", pi: " << ae.prev_idx << ", pt: " << ae.prev_term
-      << ", lci: " << ae.leader_commit_idx;
-    log(node_id, tgt_node_id, s.str());
+    const auto s = fmt::format(
+      "append_entries ({}.{}, {}.{}] (term {}, commit {})",
+      ae.prev_term,
+      ae.prev_idx,
+      ae.term_of_idx,
+      ae.idx,
+      ae.term,
+      ae.leader_commit_idx);
+    log(node_id, tgt_node_id, s);
```

Comment:
Done a bunch of reformatting of these log lines, to be slightly more concise and match our (or at least my) current understanding of the messages. In particular, AppendEntries contains a non-inclusive _previous_ term + index (TxID), and an inclusive _final_ term + index (TxID). I've started writing this as a range `(2.3, 5.8]`.